exec/java-exec/src/main/java/org/apache/drill/exec/expr/annotations/FunctionTemplate.java

```diff
@@ -55,12 +55,19 @@
 
   FunctionScope scope();
   NullHandling nulls() default NullHandling.INTERNAL;
+
+  /**
+   * This method tells if the number of argument(s) for a function is FIXED or VARIABLE. Since most functions expect
+   * receiving fixed number of argument(s), the default value is FIXED.
+   */
+  FunctionArgumentNumber argNumber() default FunctionArgumentNumber.FIXED;
```

Comment:
As a user, I do not see how setting argNumber to VARIABLE is conveying any meaning to Drill, since we do not support specifying variable number of arguments using Param annotation.

If the problem is with _concat_, maybe we should have special handling for certain functions, while registering in DrillFunctionRegistry, rather than through ScanResult?